client/app/scripts/liveblog-advertising/controllers/advertising.js

```diff
@@ -50,26 +59,33 @@ upload, $templateCache, freetypeService, modal) {
         $scope.advertModalActive = false;
     }
 
-    function loadAdverts() {
-        $scope.advertsLoading = true;
-        api('advertisements').query({not: {term: {deleted: true}}}).then(function(data) {
+    function loadAdverts(silent) {
+        silent = silent || false;
+        !silent ? $scope.advertsLoading = true : $scope.advertsLoading = false;
+        api('advertisements').query({where: {deleted: false}}).then(function(data) {
             $scope.adverts = data._items;
-            notify.info('Adverts loaded');
+            if (!silent) {
+                notify.info('Adverts loaded');
+            }
             $scope.advertsLoading = false;
         }, function(data) {
             $scope.advertsLoading = false;
             notify.error(gettext('There was an error getting the adverts'));
         })
     }
 
-    function handleSaveSuccess() {
+    function handleAdvertSaveSuccess() {
         notify.info(gettext('Advert saved successfully'));
         $scope.freetypeControl.reset();
         $scope.advertModalActive = false;
         $scope.dialogAdvertLoading = false;
         loadAdverts();
     }
 
+    function handleAdvertSaveError() {
+        notify.error(gettext('Something went wrong, please try again later!'), 5000)
+    }
+
     $scope.removeAdvert = function (advert, $index) {
         modal.confirm(gettext('Are you sure you want to remove this advert?')).then(function() {
             api('advertisements').save(advert, {deleted: true}).then(function(data) {
```

Comment:
I tried it, but if done so, the modal functionality breaks( you have to push the OK button a second time to close the modal, even if the action takes place, or if using catch then the notification error is shown when pressing the Cancel button)